trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-for-report-helper.ts

```diff
@@ -222,12 +227,62 @@ const mapPaymentToPaymentDetailsPaymentViewModel = (
 const getPaymentsSortedByDateReceived = (payments: Payment[]): Payment[] =>
   orderBy(payments, [({ dateReceived }) => parseISO(dateReceived).getTime()], ['asc']);
 
+/**
+ * Gets the payment id to date reconciled data sort value map
+ * @param feeRecordPaymentGroups - The fee record payment groups
+ * @returns The payment id to date reconciled data sort value map
+ */
+const getPaymentIdToDateReconciledDataSortValueMap = (feeRecordPaymentGroups: FeeRecordPaymentGroup[]): Record<number, number> => {
+  const allPaymentsWithDateReconciled = feeRecordPaymentGroups.reduce(
+    (paymentsWithDateReconciled, { paymentsReceived, dateReconciled }) => {
+      if (!paymentsReceived) {
+        return paymentsWithDateReconciled;
+      }
+      return [...paymentsWithDateReconciled, ...paymentsReceived.map((payment) => ({ ...payment, dateReconciled }))];
+    },
+    [] as (Payment & { dateReconciled?: IsoDateTimeStamp })[],
+  );
+  return orderBy(allPaymentsWithDateReconciled, [({ dateReconciled }) => (dateReconciled === undefined ? -1 : parseISO(dateReconciled).getTime())]).reduce(
+    (map, { id }, paymentIndex) => ({
+      ...map,
+      [id]: paymentIndex,
+    }),
+    {} as Record<number, number>,
+  );
+};
+
+/**
+ * Gets the formatted reconciled by user
+ * @param reconciledByUser - The reconciled by user
+ * @returns The formatted reconciled by user
+ */
+const getFormattedReconciledByUser = (reconciledByUser: { firstName: string; lastName: string } | undefined): string =>
+  reconciledByUser ? `${reconciledByUser.firstName} ${reconciledByUser.lastName}` : '-';
+
+/**
+ * Formats the date reconciled
+ *
+ * The "aaaaa'm'" selection is so we can format the date to use
+ * lowercase 'AM' or 'PM' and comes from the below link.
+ * {@link https://stackoverflow.com/questions/60728212/how-do-you-format-a-datetime-to-am-pm-without-periods-when-using-date-fns-vers}
+ * @param dateReconciled - The date reconciled
+ * @returns The formatted date
+ * @example
+ * getFormattedDateReconciled('2024-01-01T12:30:00.000'); // '1 Jan 2024 at 12:30pm'
+ * getFormattedDateReconciled('2024-01-01T11:30:00.000'); // '1 Jan 2024 at 11:30am'
+ * getFormattedDateReconciled(null); // '-'
+ */
+const getFormattedDateReconciled = (dateReconciled: IsoDateTimeStamp | undefined): string =>
+  dateReconciled ? format(new Date(dateReconciled), "d MMM yyyy 'at' hh:mmaaaaa'm'") : '-';
```

Comment:
Personally, I think using string literals makes more sense than having constants because date formatting can be very specific to the context it is used in, and it can therefore be hard to come up with a short but descriptive name for it. The benefits of potentially DRY code in the future should also be weighed up against YAGNI in terms of abstracting away implementation details by using the constant.

As there is currently no consistent pattern with respect to date formats, I recommend creating a DTFS ticket to commonise the date formatting approach rather than updating one instance of the formatting in this PR

Reply:
We're gradually using more date constants, e.g `portal/server/constants/date.ts`. Introducing new hard coded date formats is not ideal and I don't believe is a good practice - can we please look into how these new date format strings could be in constants?

Reply:
We can, but this is something that is beyond the scope of this PR and would be better served by a dedicated tech-debt ticket

Reply:
This PR introduces a new date format and is therefore in the scope of this PR? Can we please create a constant for this new date format.

Any other date formats that are not in constants will be moved to constants in a future PR.

Reply:
Have created